Unable to build on macos..

Looks like there is an issue with latest xcode as per stackoverflow

./boost/container_hash/hash.hpp:132:33: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
        struct hash_base : std::unary_function<T, std::size_t> {};
                           ~~~~~^~~~~~~~~~~~~~
                                __unary_function

--- a/cmake/ConfigureCompiler.cmake
+++ b/cmake/ConfigureCompiler.cmake
@@ -74,6 +74,11 @@ if(WIN32)
   add_definitions(-D_USE_MATH_DEFINES) # Math constants
 endif()
 
+if(APPLE)
+# Remove this after boost 1.81 or above is used
+add_definitions(-D_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION)
+endif()
+
 if (USE_CCACHE)
   find_program(CCACHE_PROGRAM "ccache" REQUIRED)
   set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_PROGRAM}")

--- a/fdbmonitor/fdbmonitor.cpp
+++ b/fdbmonitor/fdbmonitor.cpp
@@ -80,7 +80,9 @@
 #include "fdbclient/SimpleIni.h"
 #include "fdbclient/versions.h"
 
+#ifdef __linux__
 constexpr uint64_t DEFAULT_MEMORY_LIMIT = 8LL << 30;
+#endif
 constexpr double MEMORY_CHECK_INTERVAL = 2.0; // seconds
 
 #ifdef __linux__

--- a/fdbserver/tester.actor.cpp
+++ b/fdbserver/tester.actor.cpp
@@ -913,7 +913,7 @@ ACTOR Future<Void> clearData(Database cx, Optional<TenantName> defaultTenant) {
 
 			std::vector<Future<Void>> deleteFutures;
 			for (auto const& [id, entry] : tenants.results) {
-				if (entry.tenantName != defaultTenant) {
+				if (!defaultTenant.present() || entry.tenantName != defaultTenant.get()) {
 					deleteFutures.push_back(TenantAPI::deleteTenantTransaction(&tr, id));
 				}
 			}

--- a/fdbserver/workloads/DcLag.actor.cpp
+++ b/fdbserver/workloads/DcLag.actor.cpp
@@ -75,7 +75,7 @@ struct DcLagWorkload : TestWorkload {
 		std::vector<IPAddress> ips; // all remote process IPs
 		for (const auto& process : g_simulator->getAllProcesses()) {
 			const auto& ip = process->address.ip;
-			if (process->locality.dcId().present() && process->locality.dcId().get() == g_simulator->remoteDcId) {
+			if (process->locality.dcId().present() && process->locality.dcId().get() == g_simulator->remoteDcId.get()) {
 				ips.push_back(ip);
 			}
 		}

--- a/fdbserver/workloads/GcGenerations.actor.cpp
+++ b/fdbserver/workloads/GcGenerations.actor.cpp
@@ -97,7 +97,7 @@ struct GcGenerationsWorkload : TestWorkload {
 		std::vector<IPAddress> remoteIps; // all remote process IPs
 		for (const auto& process : g_simulator->getAllProcesses()) {
 			const auto& ip = process->address.ip;
-			if (process->locality.dcId().present() && process->locality.dcId().get() == g_simulator->remoteDcId &&
+			if (process->locality.dcId().present() && process->locality.dcId().get() == g_simulator->remoteDcId.get() &&
 			    !isCoordinator(coordinators, ip)) {
 				remoteIps.push_back(ip);
 			} else {

--- a/metacluster/include/metacluster/MetaclusterConsistency.actor.h
+++ b/metacluster/include/metacluster/MetaclusterConsistency.actor.h
@@ -199,11 +199,11 @@ private:
 		ASSERT(data.metaclusterRegistration.get().id == clusterMetadata.entry.id);
 
 		if (data.tenantData.lastTenantId >= 0) {
-			ASSERT_EQ(TenantAPI::getTenantIdPrefix(data.tenantData.lastTenantId), managementData.tenantIdPrefix);
+			ASSERT_EQ(TenantAPI::getTenantIdPrefix(data.tenantData.lastTenantId), managementData.tenantIdPrefix.get());
 			ASSERT_LE(data.tenantData.lastTenantId, managementData.tenantData.lastTenantId);
 		} else {
 			for (auto const& [id, tenant] : data.tenantData.tenantMap) {
-				ASSERT_NE(TenantAPI::getTenantIdPrefix(id), managementData.tenantIdPrefix);
+				ASSERT_NE(TenantAPI::getTenantIdPrefix(id), managementData.tenantIdPrefix.get());
 			}
 		}
 
@@ -243,7 +243,8 @@ private:
 				ASSERT_EQ(metaclusterEntry.tenantState, TenantState::READY);
 				ASSERT(entry.tenantName == metaclusterEntry.tenantName);
 			} else if (entry.tenantName != metaclusterEntry.tenantName) {
-				ASSERT(entry.tenantName == metaclusterEntry.renameDestination);
+				ASSERT(metaclusterEntry.renameDestination.present() &&
+				       entry.tenantName == metaclusterEntry.renameDestination.get());
 			}
 			if (metaclusterEntry.tenantState != TenantState::UPDATING_CONFIGURATION &&
 			    metaclusterEntry.tenantState != TenantState::REMOVING) {

